autoprotocol/protocol.py

```diff
@@ -5080,51 +5081,62 @@ def provision(self, resource_id, dests, volumes):
-        if not isinstance(volumes, list):
-            volumes = [Unit(volumes)] * len(dests)
+        if not isinstance(amounts, list):
+            amounts = [Unit(amounts)] * len(dests)
         else:
-            if len(volumes) != len(dests):
+            if len(amounts) != len(dests):
                 raise RuntimeError(
                     "To provision a resource into multiple "
-                    "destinations with multiple volumes, the  "
-                    "list of volumes must correspond with the "
+                    "destinations with multiple volumes/masses, the  "
+                    "list of volumes/masses must correspond with the "
                     "destinations in length and in order."
                 )
-            volumes = [Unit(v) for v in volumes]
-        for v in volumes:
-            if not isinstance(v, (str, Unit)):
-                raise TypeError("Volume must be a string or Unit.")
+            amounts = [Unit(v) for v in amounts]
+        unique_measure_modes = set()
+        for amount in amounts:
+            if not isinstance(amount, (str, Unit)):
+                raise TypeError("Volume/Mass must be a string or Unit.")
+            if amount.unit in VOLUMETRIC_UNITS:
+                unique_measure_modes.add("volume")
+            elif amount.unit in MASS_UNITS:
+                unique_measure_modes.add("mass")
+
+        if len(unique_measure_modes) != 1:
+            raise ValueError("Received amounts with more than one measurement mode")
+        measurement_mode = unique_measure_modes.pop()
         provision_instructions_to_return: Provision = []
-        for d, v in zip(dests, volumes):
-            d_max_vol = d.container.container_type.true_max_vol_ul
-            if v > d_max_vol:
-                raise ValueError(
-                    f"The volume you are trying to provision ({v}) exceeds the "
-                    f"maximum capacity of this well ({d_max_vol})."
-                )
-            self._remove_cover(d.container, "provision")
-            dest_group = []
-            if v > Unit(900, "microliter"):
-                diff = v - Unit(900, "microliter")
-                provision_instructions_to_return.append(
-                    self.provision(resource_id, d, Unit(900, "microliter"))
-                )
-                while diff > Unit(0.0, "microliter"):
-                    provision_instructions_to_return.append(
-                        self.provision(resource_id, d, diff)
-                    )
-                    diff -= diff
-                continue
+
+        for d, amount in zip(dests, amounts):
 
             xfer = {}
             xfer["well"] = d
-            xfer["volume"] = v
+            xfer[measurement_mode] = amount
 
-            if d.volume:
-                d.volume += v
-            else:
-                d.set_volume(v)
-            dest_group.append(xfer)
+            if measurement_mode == "volume":
```

Comment:
We briefly talked about the possibility of doing the max_vol check by converting with density... Probably more of a nice-to-have

Reply:
Since there is no consensus yet on this, I will wait on product owner to conform on the requirement. May be I will reach out to apps team try to get their input. But I prefer to have PO take final call and put it in the issue.

Reply:
Based on [comment](https://strateos.atlassian.net/browse/SEA-240?focusedCommentId=16025)  from product, we are not adding any checks or logic as part of this US implementation